### PR TITLE
Add annotations to model properties

### PR DESCRIPTION
**Pull request by MikeWNelson**
Use properties from schema to generate DataAnnotations attributes on a model.

### CODE REVIEW

File: lib/PrincipleStudios.OpenApi.CSharp/Templates/objectmodel.handlebars [hunk 1]
```diff
@@ -14,6 +14,11 @@ namespace {{this.PackageName}}
         [property: global::System.Text.Json.Serialization.JsonPropertyName("{{BaseName}}")]{{#if Required
             }}[property: global::System.Text.Json.Serialization.JsonRequiredAttribute]{{else
             }}[property: global::System.Text.Json.Serialization.JsonIgnore(Condition = global::System.Text.Json.Serialization.JsonIgnoreCondition.WhenWritingDefault)]{{/if}}{{!
+            }}{{#if Required}}[global::System.ComponentModel.DataAnnotations.Required]{{/if}}{{!
```
Comment (MikeWNelson) [inline lib/PrincipleStudios.OpenApi.CSharp/Templates/objectmodel.handlebars]: @mdekrey This Required attribute is breaking some of the tests, specifically around the the 'nullable' properties marked as required in the schema (at least from what I can tell). ModelState is invalid.  Any advice on this? Thanks.

Comment (mdekrey) [reply lib/PrincipleStudios.OpenApi.CSharp/Templates/objectmodel.handlebars]: .NET's `RequiredAttribute` has slightly different meanings than OpenAPI's `required` array - empty strings are considered missing to .NET. Let me take a look.

Comment (mdekrey) [reply lib/PrincipleStudios.OpenApi.CSharp/Templates/objectmodel.handlebars]: ~I'm not seeing any failures locally. We should try to pair later today!~ I'm able to reproduce these; the `.TestApp` didn't regenerate the files.

Comment (mdekrey) [reply lib/PrincipleStudios.OpenApi.CSharp/Templates/objectmodel.handlebars]: Yeah, OpenAPI's `required` should be handled only through the `JsonRequiredAttribute`, which will fail the deserialization if it's not provided. We should not be using `global::System.ComponentModel.DataAnnotations.Required` here.

```suggestion
```

File: lib/PrincipleStudios.OpenApi.CSharp/CSharpSchemaSourceResolver.cs [hunk 2]
```diff
@@ -227,7 +227,12 @@ private Func<Templates.ObjectModel> ToObjectModel(string className, OpenApiSchem
 												   IsContainer: resolved.isEnumerable,
 												   Name: CSharpNaming.ToPropertyName(entry.Key, options.ReservedIdentifiers("object", className)),
 												   Required: req,
-												   Optional: !req && !objectModel.LegacyOptionalBehavior
+												   Optional: !req && !objectModel.LegacyOptionalBehavior,
+												   Pattern: entry.Value.Pattern,
+												   MinLength: entry.Value.MinLength,
+												   MaxLength: entry.Value.MaxLength,
+												   Minimum: entry.Value.Minimum,
+												   Maximum: entry.Value.Maximum
```
Comment (mdekrey) [inline lib/PrincipleStudios.OpenApi.CSharp/CSharpSchemaSourceResolver.cs:235]: These are useful additions, thanks!

File: lib/TestApp/Annotations/Controllers.cs [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+namespace PrincipleStudios.OpenApiCodegen.Server.Mvc.TestApp.Annotations
+{
+	public class DogController : DogControllerBase
+	{
+		protected override Task<AddDogActionResult> AddDog(Dog addDogBody)
+		{
+			this.DelegateRequest(addDogBody);
+			return this.DelegateResponse<AddDogActionResult>();
+		}
+	}
```
Comment (mdekrey) [inline lib/TestApp/Annotations/Controllers.cs:12]: Looks like a good implementation, but I don't think I see any tests that calls this to ensure the annotations work the way we expect. Can you write a few positive/negative tests for those annotations?

File: schemas/annotations.yaml [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+        hairColor:
+          type: string
+          pattern: "^[a-zA-Z-0-9 ]{3,100}|null$"
+        eyeColor:
+          type: string
+          pattern: "^[a-zA-Z-0-9 ]{3,100}|null$"
+          minLength: 1
+          maxLength: 50
+          minimum: 0
+          maximum: 10
```
Comment (mdekrey) [inline schemas/annotations.yaml]: `|null` is confusing here - the `hairColor` should be able to be omitted, and `null` would still match the other half of the pattern.

Also, what would the competing minimum/minLength do here? I feel like these are invalid annotation combinations for testing.

File: lib/TestApp/Annotations/Controllers.cs (hunk 3 above)
Comment (MikeWNelson) [reply lib/TestApp/Annotations/Controllers.cs:12]: Tests added.

File: schemas/annotations.yaml (hunk 4 above)
Comment (MikeWNelson) [reply schemas/annotations.yaml]: Schema updated

**Review by mdekrey — APPROVED** [review]
Looks great! I updated the tests to follow Xunit practices a bit better, and bumped the version number so when it's merged, a new version will be available shortly thereafter!

**Pull request merged**